src/inject.js

```diff
@@ -45,12 +44,21 @@ export default function injectIntl(WrappedComponent, options = {}) {
     }
 
     render() {
-      return (
-        <WrappedComponent
-          {...this.props}
-          {...{[intlPropName]: this.context.intl}}
-          ref={withRef ? 'wrappedInstance' : null}
-        />
+      if (!this.context.intl) {
+        const DeepWrappedComponent = injectIntl(WrappedComponent);
```

Comment:
We were using an approach similar to this in our application code and found that the component is recreated on each render if the intl context is missing.

I solved this by only creating the `DeepWrappedComponent` once and storing it on the `InjectIntl` instance.